### PR TITLE
add tests for better coverage of pcdm objects

### PR DESCRIPTION
**Pull request by elrayle**
This primarily extends pcdm object tests to make them consistent with testing for pcdm collection.  This also added object hasMember object in the same way that collection hasMember collection.

### CODE REVIEW

File: lib/hydra/pcdm/models/concerns/object_behavior.rb [hunk 1]
```diff
@@ -60,14 +57,36 @@ def objects
...
 
+    def object_ancestor? objects
+      objects.each do |check|
+        return true if check.id == self.id
+        return true if ancestor?(check)
+      end
+      false
+    end
+
+    def ancestor? object
```
Comment (jcoyne) [inline lib/hydra/pcdm/models/concerns/object_behavior.rb]: Please add comments about what this method does.

File: lib/hydra/pcdm/models/concerns/object_behavior.rb [hunk 2]
```diff
@@ -60,14 +57,36 @@ def objects
       members.to_a.select { |m| Hydra::PCDM.object? m }
     end
 
+    def object_ancestor? objects
```
Comment (jcoyne) [inline lib/hydra/pcdm/models/concerns/object_behavior.rb]: Please add a comment about what this method does.

File: lib/hydra/pcdm/models/concerns/object_behavior.rb (hunk 1 above)
Comment (jcoyne) [inline lib/hydra/pcdm/models/concerns/object_behavior.rb]: Should this method be private?

File: lib/hydra/pcdm/models/concerns/object_behavior.rb [hunk 3]
```diff
@@ -60,14 +57,36 @@ def objects
...
+      objects.each do |check|
```
Comment (jcoyne) [inline lib/hydra/pcdm/models/concerns/object_behavior.rb]: A more concise way to write this is:
```objects.any? { |check| check.id == id || ancestor?(check)}```

File: spec/hydra/pcdm/models/object_spec.rb [hunk 4]
```diff
@@ -2,70 +2,150 @@
...
+          end
+        end
+      end
     end
   end
 
   describe '#objects' do
     it 'should return empty array when no members' do
-      object1 = Hydra::PCDM::Object.new
+      object1.save
```
Comment (jcoyne) [inline spec/hydra/pcdm/models/object_spec.rb]: why are you saving here? Save is a slow operation.

**Comment by elrayle** [comment]
The code in question implements object hasMember object.  It is directly modeled off the same code for collection hasMember collection in collection_behavior.

https://github.com/projecthydra-labs/hydra-pcdm/blob/master/lib/hydra/pcdm/models/concerns/collection_behavior.rb

methods: 
    def collection_ancestor? collections
    def ancestor? collection

If changes need to happen to make this function better, then they need to happen there as well.

**Pull request merged**